### PR TITLE
ddoc: Better define section names

### PR DESCRIPTION
**Pull request by jblachly**
Clarify difference in section and section name. Also explicilty describe that section names may not contain spaces, but that underscores will be replaced with spaces.

### CODE REVIEW

File: spec/ddoc.dd [hunk 1]
```diff
@@ -199,11 +199,17 @@ $(H3 $(LNAME2 sections, Sections))
 
 $(P
 The document comment is a series of $(I Section)s.
-A $(I Section) is a name that is the first non-blank character on
+A $(I Section name) is a name that is the first non-blank character on
 a line immediately followed by a ':'. This name forms the section name.
 The section name is not case sensitive.
```
Comment (Hackerpilot) [inline spec/ddoc.dd:204]: This seems to not actually be true. I just tried writing `section_name:` vs `Section_name:`, and only the upper-case version is treated as a section name by ddoc in DMD 2.093.0.

Comment (jblachly) [reply spec/ddoc.dd:204]: ddoc, while a fantastic idea, leaves a lot to be desired. At least we are moving toward markdown.

File: spec/ddoc.dd [hunk 2]
```diff
@@ -199,11 +199,17 @@ $(H3 $(LNAME2 sections, Sections))
...
 )
 
+$(P
+Section names do not contain whitespace. Underscores will be replaced with spaces.
+For example, a section entitled 'New API' would be written with having its first line
+'New_API:'
```
Comment (mdparker) [inline spec/ddoc.dd:210]: "written with having its first line" is rather awkward. And it isn't accurate. The section name doesn't need to be the first line, but at the front of the first line. So something like:

"For example, the first line of a section titled 'New API' would begin with 'New_API:'."

**Review by mdparker — CHANGES_REQUESTED** [review]

File: spec/ddoc.dd (hunk 2 above)
Comment (jblachly) [reply spec/ddoc.dd:210]: Strike "with" and change "would" to "could" makes it both grammatical and semantically accurate.

**Comment by jblachly** [comment]
The response to this PR is a symptom of something bigger

**Pull request closed without merging**